### ISSUE TITLE
saturation_preconditioner always return -1 for cubics

### ISSUE THREAD

**Issue by JonWel** [CoolProp/CoolProp#1218]
I think that could be a reason for the cubic saturation evaluation not being as quick as expected for mixtures solved with cubic.

The (saturation_preconditioner)[https://github.com/CoolProp/CoolProp/blob/master/src/Backends/Helmholtz/VLERoutines.h#L160] should provide a guess for the pressure or for the temperature, and -1 is certainly not the most helping guess value (not physical values).
It always return -1 as it uses `components` which is not a compatible variable in cubic's case.

I guess that force more iterations in the solver.

**Comment by ibell** [comment]
I think this is a good question.  For QT inputs, it actually doesn't matter because once you arrive in Wilson_preconditioner, you ignore the guess completely because you can get an explicit solution: see http://www.coolprop.dreamhosters.com/binaries/sphinx/_static/doxygen/html/namespace_cool_prop_1_1_saturation_solvers.html#a3adeb9a36ecdf789accc84d113047078 (docs are currently broken).  For p imposed, some better guess is needed

**Comment by JonWel** [comment]
Yes, I saw a bit later that this guess was not used for saturation QT.
Shouldn't we move the logic (`if (input_type == imposed_T && (std::abs(beta) < 1e-12 || std::abs(beta-1) < 1e-12))`) at a higher level (`QT_flash` and `PQ_flash`), so we execute only what is needed?
(would slightly benefit QT, but it's not a big overhead so no hurry).

For p imposed, how can we have a compatible `saturation_preconditioner`?

**Comment by ibell** [comment]
Closing as part of an issue-tracker cleanup — this issue has had no activity in over two years.

*To be transparent: this comment and the close action are being performed by Claude (Anthropic's AI coding assistant), running on behalf of @ibell. Ian reviewed and approved the bulk-close criteria; I am the one executing the action.*

If you would like this issue reconsidered, please feel free to **re-open it** — a fresh comment with current behaviour against the latest CoolProp release is the most useful way to revive the discussion.

— Claude (acting for @ibell)
